libs/core/async_sycl/include/hpx/async_sycl/sycl_executor.hpp

```diff
@@ -0,0 +1,324 @@
+static_assert(std::is_same<std::decay_t<Executor>,
+    hpx::sycl::experimental::sycl_executor>::value);
+// Use the same async_dispatch than the normal async otherwise
+return detail::async_dispatch<typename std::decay<Executor>::type>::call(
+    HPX_FORWARD(Executor, exec),
+    HPX_FORWARD(
+        hpx::sycl::experimental::sycl_executor::queue_function_ptr_t<Ts...>, f),
+    HPX_FORWARD(Ts, ts)...);
+}
+
+/// hpx::apply overload for launching sycl queue member functions with an
+/// sycl executor
+template <typename Executor, typename... Ts>
+HPX_FORCEINLINE bool apply(Executor&& exec,
+    hpx::sycl::experimental::sycl_executor::queue_function_ptr_t<Ts...>&& f,
+    Ts&&... ts)
+{
+// Make sure we only use this for sycl executors
+static_assert(std::is_same<std::decay_t<Executor>,
+    hpx::sycl::experimental::sycl_executor>::value);
+// Use the same apply_dispatch than the normal apply otherwise
+return detail::post_dispatch<typename std::decay<Executor>::type>::call(
+    HPX_FORWARD(Executor, exec),
+    HPX_FORWARD(
+        hpx::sycl::experimental::sycl_executor::queue_function_ptr_t<Ts...>, f),
+    HPX_FORWARD(Ts, ts)...);
+}
+
+//#if defined(__LIBSYCL_MAJOR_VERSION) && defined(__LIBSYCL_MINOR_VERSION)
+#if defined(__INTEL_LLVM_COMPILER) ||                                          \
+    (defined(__clang__) && defined(SYCL_IMPLEMENTATION_ONEAPI)) 
+/// hpx::async overload for launching sycl queue member functions with an
+/// sycl executor and code location ptrs
+template <typename Executor, typename Ts>
+HPX_FORCEINLINE decltype(auto) async(Executor&& exec,
+    hpx::sycl::experimental::sycl_executor::queue_function_code_loc_ptr_t<Ts>&&
+        f,
+    Ts&& ts)
+{
+// Make sure we only use this for sycl executors
+static_assert(std::is_same<std::decay_t<Executor>,
+    hpx::sycl::experimental::sycl_executor>::value);
+// Use the same async_dispatch than the normal async otherwise
+return detail::async_dispatch<typename std::decay<Executor>::type>::call(
+    HPX_FORWARD(Executor, exec),
+    HPX_FORWARD(
+        hpx::sycl::experimental::sycl_executor::queue_function_code_loc_ptr_t<Ts>, f),
+    HPX_FORWARD(Ts, ts));
+}
+
+/// hpx::apply overload for launching sycl queue member functions with an
+/// sycl executor and code location ptrs
+template <typename Executor, typename Ts>
+HPX_FORCEINLINE bool apply(Executor&& exec,
+    hpx::sycl::experimental::sycl_executor::queue_function_code_loc_ptr_t<Ts>&&
+        f,
+    Ts&& ts)
+{
+// Make sure we only use this for sycl executors
+static_assert(std::is_same<std::decay_t<Executor>,
+    hpx::sycl::experimental::sycl_executor>::value);
+// Use the same apply_dispatch than the normal apply otherwise
+return detail::post_dispatch<typename std::decay<Executor>::type>::call(
+    HPX_FORWARD(Executor, exec),
+    HPX_FORWARD(
+        hpx::sycl::experimental::sycl_executor::queue_function_code_loc_ptr_t<Ts>, f),
+    HPX_FORWARD(Ts, ts));
+}
+#endif
+}    // namespace hpx
```

Comment:
Something is off with the formatting in this file, does it need processing with clang-format?